src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -99,22 +94,21 @@ public RuntimeConfig GetConfig()
     /// </summary>
     private bool TrySetupConfigFileWatcher()
     {
-        if (!_hasAttemptedFileWatcherConfiguration)
+        if (!IsLateConfigured && _runtimeConfig is not null && _runtimeConfig.IsDevelopmentMode())
         {
-            if (!IsLateConfigured && _runtimeConfig is not null && _runtimeConfig.IsDevelopmentMode())
+            try
             {
-                try
-                {
-                    FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)ConfigLoader;
-                    _configFileWatcher = new(this, loader.GetConfigDirectoryName(), loader.GetConfigFileName());
-                    _hasAttemptedFileWatcherConfiguration = true;
-                    return true;
-                }
-                catch (Exception)
-                {
-                    return false;
-                }
+                FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)ConfigLoader;
+                _configFileWatcher = new(this, loader.GetConfigDirectoryName(), loader.GetConfigFileName());
             }
+            catch (Exception ex)
+            {
+                // Need to remove the dependencies in startup on the RuntimeConfigProvider
+                // before we can have an ILogger here.
+                Console.WriteLine($"Attempt to configure config file watcher for hot reload failed due to: {ex.Message}.");
```

Comment:
Dont we have access to logger?

Reply:
nvm see the comment now

Reply:
before this line of code is executed, we will :)

Reply:
i need to refactor and resolve some dependencies on the runtimeconfig provider in startup to get ILogger here.